spinta/datasets/backends/sql/commands/read.py

```diff
@@ -105,10 +102,22 @@ def getall(
                     if isinstance(sel.prop.dtype, PrimaryKey):
                         val = keymap.encode(sel.prop.model.model_type(), val)
                     elif isinstance(sel.prop.dtype, Ref):
-                        val = keymap.encode(sel.prop.dtype.model.model_type(), val)
+                        if sel.prop.dtype.model.external.pkeys == sel.prop.dtype.refprops or len(
+                            sel.prop.dtype.refprops) > len(sel.prop.dtype.model.external.pkeys):
+                            val = keymap.encode(sel.prop.dtype.model.name, val)
+                        else:
+                            table = backend.get_table(model, str(sel.prop.dtype.model.external.name))
+                            pk = conn.execute(_get_params_values(table, val)).scalar()
```

Comment:
This generates multiple queries and we don't want to do this, because it slows down read operation significantly, `conn.execute` is very expensive. We have to retrieve primary key from main query, by doing a join in the main query.